### PR TITLE
Migrate MapboxTestHost to Firebase device lab testing

### PR DESCRIPTION
**Pull request by OdNairy**
Migrate MapboxTestHost to Firebase device lab testing

## Pull request checklist:
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Describe the changes in this PR, especially public API changes.
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [x] Review and agree to the Contributor License Agreement ([CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement)).

Fixes https://github.com/mapbox/mapbox-maps-ios-internal/issues/1010

### CODE REVIEW

**Review by macdrevx — COMMENTED** [review]
Can we preserve the ability to trigger running device tests from a branch? Previously it worked by using a key phrase in the commit message.

**Comment by OdNairy** [comment]
> Can we preserve the ability to trigger running device tests from a branch? Previously it worked by using a key phrase in the commit message.

I would recover that option in following PRs

File: .circleci/config.yml [hunk 1]
```diff
@@ -418,13 +395,50 @@ jobs:
...
+      xcode: 13.2.1
+    steps:
+      - checkout
+      - configure-environment
+      - setup-google-cloud-account
+      - attach_workspace:
+          at: output
+      - install-gcloud
+      - login-google-cloud-platform
+      # - run: bundle install
```
Comment (macdrevx) [inline .circleci/config.yml:420]: Can we delete this?

File: .circleci/config.yml [hunk 2]
```diff
@@ -124,19 +114,6 @@ workflows:
           filters:
             branches:
               ignore: main
-      # On main
-      - run-tests-on-devices:
```
Comment (macdrevx) [inline .circleci/config.yml:128]: Can this job definition be deleted now?

File: .fastlane/Fastfile [hunk 3]
```diff
@@ -96,14 +96,44 @@ platform :ios do
...
+      gcp_project: 'mapbox-maps-ios-dc24c', # Your Google Cloud project name
+      devices: [                          # Device(s) to run tests on
+        {
+          ios_model_id: 'iphone11',        # Device model ID, see gcloud command above
+          ios_version_id: '13.6',         # iOS version ID, see gcloud command above
+        }
+      ],
+      timeout_sec: 20*60,
+      skip_validation: true,
+      # result_storage: storage_path
```
Comment (macdrevx) [inline .fastlane/Fastfile:135]: Some commented out code here. Can it be cleaned up?

File: .circleci/config.yml (hunk 2 above)
Comment (OdNairy) [reply .circleci/config.yml:128]: I'm going to do a huge clean up of all DeviceFarm related scripts/jobs/makefile rules after I finish building the same behaviour. Currently we are missing trigger-by-commit in non-main branches

File: .fastlane/Fastfile (hunk 3 above)
Comment (OdNairy) [reply .fastlane/Fastfile:135]: It would be on later steps

**Review by macdrevx — APPROVED** [review]

**Pull request merged**